### PR TITLE
chore: use colors from theme for notification's severity

### PR DESCRIPTION
**Pull request by mareklibra**
## Hey, I just made a Pull Request!

Colors of severity icons are based on theme.

Follow-up for https://github.com/backstage/backstage/pull/23371#discussion_r1524651872

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by mareklibra** [comment]
Screenshot from 2024-03-14 12-50-46

File: plugins/notifications/src/api/NotificationsClient.ts [hunk 1]
```diff
@@ -68,7 +68,7 @@ export class NotificationsClient implements NotificationsApi {
       queryString.append('created_after', options.createdAfter.toISOString());
     }
     if (options?.minimumSeverity !== undefined) {
-      queryString.append('minimal_severity', options.minimumSeverity);
+      queryString.append('minimumSeverity', options.minimumSeverity);
```
Comment (Rugvip) [inline plugins/notifications/src/api/NotificationsClient.ts:71]: There are a couple other above as well, best upgrade them all at once

**Review by Rugvip — CHANGES_REQUESTED** [review]
Nice! 👍 

Icons update looks good, for the casing I think we should apply it more broadly

**Comment by mareklibra** [comment]
not stale, it just needs 2nd review round.
Cc @Rugvip

**Comment by mareklibra** [comment]
not stale

File: plugins/notifications/src/components/NotificationsTable/SeverityIcon.tsx [hunk 2]
```diff
@@ -19,21 +19,39 @@ import NormalIcon from '@material-ui/icons/CheckOutlined';
...
+    color: theme.palette.error.main,
+  },
+  high: {
+    color: theme.palette.warning.main,
+  },
+  normal: {
+    color: theme.palette.success.main,
+  },
+  low: {
+    color: theme.palette.primary.main,
```
Comment (Rugvip) [inline plugins/notifications/src/components/NotificationsTable/SeverityIcon.tsx:35]: Realized this isn't a very good color to use, and some of the other ones too. You'll easily end up in a situation where integrators theme their instance in a way that breaks this. Perhaps the colors from `palette.status.*` work better?

**Review by Rugvip — CHANGES_REQUESTED** [review]
Alright got back to this one, added changesets, but I also realized the colors picked in the palette are a bit fragile.

File: plugins/notifications/src/components/NotificationsTable/SeverityIcon.tsx (hunk 2 above)
Comment (mareklibra) [reply plugins/notifications/src/components/NotificationsTable/SeverityIcon.tsx]: Done. The screenshot above is updated to conform the recent state.

**Comment by mareklibra** [comment]
not stale.
Cc @Rugvip

**Review by Rugvip — APPROVED** [review]
Sweet, thank you! 👍

**Pull request merged**